### ISSUE TITLE
Consider integrating guetzli

### ISSUE THREAD

**Issue by axyz** [axyz/optimal-image#27]
https://github.com/google/guetzli

Guetzli promise higher compression level with less quality decrease at the cost of a pretty expensive computation. As optimal-image is not intended to be used for on the fly generation, but more in an asynchronous batch scenario, highly expensive compression encoders may be a good fit.

**Comment by addityasingh** [comment]
Should it be a binding with the  as first step to compare and then the actual encoder in RUST? Looking at the C++ implementation the latter would be quite some feat

**Comment by axyz** [comment]
yeah, all native bindings fro third party encoders. only jpeg is currently using a rust library, but eventually can also add more jpeg encoders (e.g. mozjpeg) through their bindings, as outputs can vary quite a lot with different implementations.
